lib/models/device.ts

```diff
@@ -3436,6 +3436,125 @@ const getDeviceModel = function (
 				});
 			},
 		}),
+
+		/**
+		 * @namespace balena.models.device.history
+		 * @memberof balena.models.device
+		 */
+		history: addCallbackSupportToModule({
+			/**
+			 * @summary Get all history entries for a device
+			 * @name getAllByDevice
+			 * @public
+			 * @function
+			 * @memberof balena.models.device.history
+			 *
+			 * @param {String|Number} uuidOrId - device uuid (32 / 62 digits string) or id (number)
+			 * @param {Object} [options={}] - extra pine options to use
+			 * @fulfil {Object[]} - device history
+			 * @returns {Promise}
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByDevice('7cf02a687b74206f92cb455969cf8e98').then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByDevice(999999).then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByDevice('7cf02a687b74206f92cb455969cf8e98', function(error, entries) {
+			 * 	if (error) throw error;
+			 * 	console.log(entries)
+			 * });
+			 */
+			async getAllByDevice(
+				uuidOrId: string | number,
+				options?: PineOptions<DeviceHistory>,
+			): Promise<DeviceHistory[]> {
+				if (options == null) {
+					options = {};
+				}
+
+				let $filter = {};
+
+				if (isId(uuidOrId)) {
+					$filter = { tracks__device: uuidOrId };
+				} else if (isUuid(uuidOrId)) {
+					$filter = { uuid: uuidOrId };
+				} else {
+					throw new errors.BalenaInvalidParameterError('uuidOrId', uuidOrId);
+				}
+
+				return await pine.get<DeviceHistory>({
+					resource: 'device_history',
+					options: mergePineOptions(
+						{
+							$filter,
+						},
+						options,
+					),
+				});
+			},
+
+			/**
+			 * @summary Get all device history entries by application
+			 * @name getAllByApplication
+			 * @public
+			 * @function
+			 * @memberof balena.models.device.history
+			 *
+			 * @param {String|Number} slugOrUuidOrId - application slug (string), uuid (string) or id (number)
+			 * @param {Object} [options={}] - extra pine options to use
+			 * @fulfil {Object[]} - device history
+			 * @returns {Promise}
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByApplication('myorganization/myapp').then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByApplication(999999).then(function(entries) {
+			 * 	console.log(entries);
+			 * });
+			 *
+			 * @example
+			 * balena.models.device.history.getAllByApplication('myorganization/myapp', function(error, entries) {
+			 * 	if (error) throw error;
+			 * 	console.log(entries)
+			 * });
+			 */
+			async getAllByApplication(
+				slugOrUuidOrId: string | number,
+				options?: PineOptions<DeviceHistory>,
+			): Promise<DeviceHistory[]> {
+				if (options == null) {
+					options = {};
+				}
+
```

Comment:
There's no need to default this at all since `mergePineOptions` is fine with nullish being passed in (and in fact can short circuit in that case to avoid work)
```suggestion
```